paddle/fluid/framework/details/exception_holder.h

```diff
@@ -95,6 +106,12 @@ class ExceptionHolder {
     type_ = kEnforceNotMet;
   }
 
+  void Catch(const memory::allocation::BadAlloc& exp) {
+    std::lock_guard<std::mutex> lock(mu_);
+    exception_.reset(new paddle::memory::allocation::BadAlloc(exp));
```

Comment:
Better Add`if (exception_.get() == nullptr)`, do not cover the existing exception.